### PR TITLE
giflib: modernize + install binaries

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **giflib/5.x**

- modernize 
- build + install binaries

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/giflib/5.2.x/CMakeLists.txt [hunk 1]
```diff
@@ -19,8 +19,36 @@ if(NOT BUILD_SHARED_LIBS)
...
+    add_executable(${GIF_UTIL} "source_subfolder/${GIF_UTIL}.c")
+    target_link_libraries(${GIF_UTIL} PRIVATE ${PROJECT_NAME} giflib_util)
+endforeach()
+find_library(M_LIBRARY NAMES m)
+if(M_LIBRARY)
+    target_link_libraries(gifclrmp PRIVATE ${M_LIBRARY})
+endif()
 
-install(TARGETS ${PROJECT_NAME}
+install(TARGETS ${PROJECT_NAME} ${GIF_UTILS}
```
Comment (SpaceIm) [inline recipes/giflib/5.2.x/CMakeLists.txt:51]: it will break iOS without BUNDLE

File: recipes/giflib/5.2.x/CMakeLists.txt [hunk 2]
```diff
@@ -19,8 +19,36 @@ if(NOT BUILD_SHARED_LIBS)
...
         ARCHIVE DESTINATION "lib"
         LIBRARY DESTINATION "lib"
         RUNTIME DESTINATION "bin")
```
Comment (SpaceIm) [inline recipes/giflib/5.2.x/CMakeLists.txt:54]:
```suggestion
        RUNTIME DESTINATION "bin"
        BUNDLE DESTINATION "bin")
```

to not break iOS

File: recipes/giflib/5.2.x/CMakeLists.txt [hunk 3]
```diff
@@ -19,8 +19,36 @@ if(NOT BUILD_SHARED_LIBS)
 endif()
 
 add_library(${PROJECT_NAME} ${SOURCE_FILES})
+if(WIN32 AND BUILD_SHARED_LIBS)
+    set_property(TARGET ${PROJECT_NAME} PROPERTY PREFIX "")
+endif()
```
Comment (SpaceIm) [inline recipes/giflib/5.2.x/CMakeLists.txt:24]: to create `gif.dll` for MinGW?

Comment (madebr) [reply recipes/giflib/5.2.x/CMakeLists.txt:24]: Yes, probably unnecessary but that will let mingw and MSVC create the same libraries.

Comment (SpaceIm) [reply recipes/giflib/5.2.x/CMakeLists.txt:24]: Yes, I still don't understand why it's not the default in CMake.

File: recipes/giflib/5.2.x/conanfile.py [hunk 4]
```diff
@@ -50,19 +57,21 @@ def _patch_sources(self):
...
+        self.copy("COPYING", src=self._source_subfolder, dst="licenses")
         cmake = self._configure_cmake()
         cmake.install()
 
-        self.copy("COPYING", src=self._source_subfolder, dst="licenses")
-
     def package_info(self):
         self.cpp_info.names["cmake_find_package"] = "GIF"
         self.cpp_info.names["cmake_find_package_multi"] = "GIF"
         self.cpp_info.libs = tools.collect_libs(self)
```
Comment (SpaceIm) [inline recipes/giflib/5.2.x/conanfile.py:71]:
```suggestion
        self.cpp_info.libs = ["gif"]
```

**Review by SpaceIm — DISMISSED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**